### ISSUE TITLE
RedundantSafeNavigation + InferNonNilReceiver vs `it`

### ISSUE THREAD

**Issue by zverok** [rubocop/rubocop#15517]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Lint/RedundantSafeNavigation

### Steps to reproduce the problem

`test.rb`:
```ruby
values = [['test', nil]]

p(values.map { it.map { it&.upcase} })
```

### Expected behavior

No offenses

### Actual behavior

```
test.rb:3:27: W: [Correctable] Lint/RedundantSafeNavigation: Redundant safe navigation on non-nil receiver (detected by analyzing previous code/method invocations).
p(values.map { it.map { it&.upcase} })
```
Correction:
```ruby
values = [['test', nil]]

p(values.map { it.map { it.upcase } })
```

The code is now invalid and will fail with 
```
undefined method 'upcase' for nil (NoMethodError)
```

### RuboCop configuration

```yaml
Lint/RedundantSafeNavigation:
  InferNonNilReceiver: true
```

### RuboCop version

```shell
$ rubocop -V
1.88.2 (using Parser 3.3.12.0, rubocop-ast 1.50.0, analyzing as Ruby 2.7, running on ruby 4.0.3) [x86_64-linux]
```

**Comment by zverok** [comment]
Actually, it is not necessary `it`. This code (while questionable in style, yet working) will be corrected too:

```ruby
values = [['test', nil]]

p(values.map { |val| val.map { |val| val&.upcase } })
```
After correction:
```ruby
p(values.map { |val| val.map(&:upcase) })
```
(will again fail with the same NoMethodError)

But while nesting blocks with the same name for a block variable is definitely rare, I believe that the approach with `it` (even if, again, questionable for some) will be seen in the wild.
